src/OmnichannelChatSDK.ts

```diff
@@ -923,7 +923,33 @@ class OmnichannelChatSDK {
         } else {
             await sessionInitPromise(); // Await the session initialization
         }
-        await Promise.all([messagingClientPromise(),attachmentClientPromise()]);
+
+        try {
+            await Promise.all([messagingClientPromise(),attachmentClientPromise()]);
+        } catch (error) {
+            // If conversation joining fails after conversation was created, clean up the conversation
+            if (error instanceof ChatSDKError &&
+                error.message === ChatSDKErrorName.MessagingClientConversationJoinFailure &&
+                !this.chatSDKConfig.useCreateConversation?.disable) {
+                try {
+                    const sessionCloseOptionalParams: ISessionCloseOptionalParams = {};
+                    if (this.authenticatedUserToken) {
+                        sessionCloseOptionalParams.authenticatedUserToken = this.authenticatedUserToken;
+                    }
+                    await this.OCClient.sessionClose(this.requestId, sessionCloseOptionalParams);
+                } catch (cleanupError) {
+                    // Log cleanup failure but don't change the original error being thrown
+                    const cleanupTelemetryData = {
+                        RequestId: this.requestId,
+                        ChatId: this.chatToken?.chatId as string,
+                        Event: 'ConversationCleanupFailure',
+                        ExceptionDetails: String(cleanupError)
+                    };
+                    this.telemetry?.error(cleanupTelemetryData);
```

Comment:
check on how each function ends iin case of error like ->                                         exceptionThrowers.throwMessagingClientConversationJoinFailure(error, this.scenarioMarker, TelemetryEvent.StartChat, telemetryData);

Reply:
you are missing to add a log entry in changelog.md , under [Unreleased] and under ### Fixed , with brief summary of the fix